backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/HttpAsyncClientBuilderConfigurer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.client;
+
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * Allows fine tuning of the Apache HTTP Client.
+ * <p>
+ * The HttpAsyncClientBuilderConfigurer implementation will be given access to the HTTP client builder
+ * on startup.
+ */
+public interface HttpAsyncClientBuilderConfigurer {
+
+	/**
+	 * Configure the HTTP Client.
+	 * <p>
+	 * This method is called once for every configurer, each time an Elasticsearch client is set up.
+	 *
+	 * @param builder An Apache HTTP client builder allowing custom settings.
+	 */
+	void configure(HttpAsyncClientBuilder builder);
```

Comment:
Actually I didn't renamed `ElasticsearchHttpClientConfigurer`, I left it to the SPIs, creating a new interface with a reduced-scope for to expose as API.

The reason because I did that is that I didn't want to expose to API stuff such as `BeanResolver` or ConfigurationPropertySource.
So basically I reduced the scope of the exposed stuff...
to only `HttpAsyncClientBuilder`.